enhancements/subscription-content/subscription-injection.md

```diff
@@ -0,0 +1,663 @@
+
+```yaml
+kind: Deployment
+apiVersion: v1
+metadata:
+  name: my-rhel-deployment
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: "rhel-deployment"
+  template:
+    metadata:
+      annotations:
+        "subscription.openshift.io/inject-cluster-bundle": cluster
+      labels:
+        app: "rhel-deployment"
+    spec:
+      containers:
+      - name: run
+        image: "registry.redhat.io/ubi8/ubi:latest"
+        ...
+```
+
+The Subscription Injection mutating admission webhook will watch all newly created pods. The
+webhook will do the following:
+
+1. Look for the `subscription.openshift.io/inject-cluster-bundle` annotation.
+2. Read the `ClusterBundles` with names matching the annotation values. Reject admission if the
+   bundle does not exist.
+3. Read the `ClusterBundles` with the `subscription.openshift.io/always-inject: "true"` label.
+   Check these bundles against the allow/deny annotations on the pod.
+4. Conduct a subject access review for the referenced `ClusterBundles` against the pod service
+   account. Return an error if the pod service account does not have `get` access to the
+   `ClusterBundle`.
+5. Verify that the `Secrets` referenced in `entitlements` across eligible `ClusterBundles` do not 
+   contain overlapping keys. Reject admission if the secrets contain overlapping keys.
+6. Add `Volumes` in the pod for the `entitlements` referenced in the bundle which sources data via
+   the projected resource CSI driver.
+7. Mount the `entitlements` volumes into `/run/secrets/etc-pki-entitlement` for each container in
+   the pod, including init and ephemeral containers.
+8. [optional] Verify that the `ConfigMaps` referenced in `yumRepositories` across elibilbe
+   `ClusterBundles` do not contain overlapping keys. Reject admission if the `ConfigMaps` contain
+   overlapping keys.
+9. [optional] Add `Volumes` in the pod for the `yumRepositories` referenced in the bundle which
+   sources data via the projected resource CSI driver.
+10. [optional] Mount the `yumRepositories` volumes into `/run/secrets` for each container in the pod,
+    including init and ephemeral containers.
+11. Record the generation of the injected bundles by adding or updating the generations JSON value
+    in the `subscription.openshift.io/bundle-generations` annotation.
+
+#### Subscription Injection Mutating Admission Webhook
+
+Per upstream docs, the mutating admission webhook will need to register itself with the cluster and
+run as a service in its own namespace [1]. The operator installing the Subscription Injection
+webhook will need to do the following:
+
+1. Create a `Deployment` that runs the webhook in a highly available mode.
+2. Create a `Service` to expose the webhook pods, with generated TLS cert/key pairs.
+3. Create a `ConfigMap` where the service serving CA can be injected.
+4. Create a `MutatingWebhookConfiguration` that fires the webhook for all created pods.
```

Comment:
this is a no-go! How will you restrict the mechanism to only pods that matter? Mutating pods is nothing we want to have for resilience reasons.

Reply:
are you aware that's exactly what Istio does to inject its sidecars?
https://istio.io/latest/docs/ops/configuration/mesh/injection-concepts/

Reply:
I am aware and it is not good for availability. If every other component thinks it is Istio ...

Reply:
At the very least this needs a risk assessment and discussion of mitigations.